.github/workflows/ci.yml

```diff
@@ -19,6 +19,10 @@ jobs:
 
     env:
       JUNIT_NAME: pytest-results-${{ matrix.python-version }}-${{ matrix.os }}
+      PYTHONDONTWRITEBYTECODE: 1
+      # make sure that no new uses of the legacy constructor are added (see also: test in
+      # test_graph.py, filterwarnings in pytest.ini)
+      PYTHONWARNINGS: "error:Constructing a StellarGraph:DeprecationWarning"
```

Comment:
I don't think this is needed now as it's in purest config?

Reply:
Sort of... it's been in the pytest config since the start; I now recall that having this as an environment variable was designed to catch problems in notebooks which don't use pytest.ini (see the description of https://github.com/stellargraph/stellargraph/pull/1027 for more discussion). And... I suspect it is the cause of the difference between Actions and Buildkite in #1711, given the build here _didn't_ have the 40k lines of logging: https://github.com/stellargraph/stellargraph/pull/1716/checks?check_run_id=798025608#step:7:91

I'll quickly remove the `PYTHONDONTWRITEBYTECODE: 1` to be sure it's `PYTHONWARNINGS` that's affecting it. (My guess is that it is, by overriding the default settings that silence the warning, or something.)

Reply:
Yep, it looks just specifying the non-default `PYTHONWARNINGS` here is enough to silence the warning-spew: https://github.com/stellargraph/stellargraph/pull/1716/checks?check_run_id=798135852#step:7:90